sdk/error.go

```diff
@@ -544,6 +544,15 @@ func NewError(httpError Error, err error) error {
 		if e, ok := err.(errorWithStack); ok {
 			e.httpError = httpError
 			return e
+		} else {
```

Comment:
This will complict with the error PR and maybe you want to do same thing

Reply:
i've just rebased my branch on master. No conflict on this.
I guess the master didn't match my needs because i have had to update unit tests to specifically check this.

here is the result on this issue (the job name is "New Job" and the action is "myAction")
<img width="390" alt="screen shot 2018-10-31 at 08 50 58" src="https://user-images.githubusercontent.com/684151/47773510-9aa89e80-dcea-11e8-9bea-3a59db2d30f4.png">

Reply:
If a understand the needs here, we want to override the origin error message for given sdk.Error. 
There can be two types of error given in NewError second's parameter, a errorWithStack and other error (sdk.Error, or lib error). What's important is to always store the second parameter as the root to preserve a valid stack trace. On a errorWithStack we can just override the http error then change the message on in with the Cause of the error.
For other types set the root and print the error as the message for http error.

Something like this ?
`
// NewError just set an error with a root cause.
func NewError(httpError Error, err error) error {
	// if the given error is a error with stack, replace the http error
	if err == nil {
    return nil
  }
	
  if e, ok := err.(errorWithStack); ok {
			e.httpError = httpError
      e.httpError.Message = Cause(err)
			return e
	}

  e:= errorWithStack{
	 	 root:      errors.WithStack(err),
		 stack:     callers(),
		 httpError: httpError,
	}
  e.httpError.Message = Cause(err)
  return err
}
`

In this case all errors from NewError will contains the root error text as message and that's what we will see on the client for or cdsctl. In this case, printing the (caused by: ...) in ExtractHTTPError will not be needed anymore.

What do you think ?

Reply:
a lot of code is using the sdk.NewError function to override the message of an sdk.Error. All the import and as-code stuff relies on this.
i don't understand all the details on the new error management, but the purpose of the related issue is to raise the detailed message to the end user.

Reply:
Ok so the master version is not good :/, because is doesn't override the message but add a caused by next the original error translated message. So I will commit on this PR to fix this and update the test case.